### DISCUSSION TITLE
Attachment digitaly signed Email

### DISCUSSION THREAD

**Discussion by martinzeifang** [ecederstrand/exchangelib#1132] [Q&A]
Hallo everyone, 

I'd like to extract an attachment from a digitally signed Email. if I do so, I just get the smine.p7m file. How can I decode the Email/Attachment to get the content? thanks everyone. 

I do just: 
```
with open(local_path, 'wb') as f:
f.write(attachment.content)
```

Thanks for the Help

Martin

**Comment by ecederstrand** [comment]
This question is not specific to exchangelib, I believe. StackOverflow has some hints on decrypting p7m files: https://security.stackexchange.com/questions/90811/decrypting-p7m-file-with-key-pem-using-openssl

**Reply by martinzeifang** [reply]
thank you! can I use exchangelib to exgtract the certificate (public key) from the email?

**Reply by martinzeifang** [reply]
one more question. if online convert the p7m file I saved with the above command I get the email not the attachment. how can I save the attachment from a signed email with exchangelib?

**Reply by ecederstrand** [reply]
Usually, a public key is not part of an encrypted message. Instead, you'll get it from some other channel you trust.

Are you sure the attachment is not contained in the encrypted file? My guess is that the p7m file contains the full MIME content.
